api/provider.go

```diff
@@ -5,5 +5,6 @@ import (
 )
 
 type Provider interface {
+	GetAllRegionsCarbonIntensity(ctx context.Context) (map[string]float64, error)
```

Comment:
This adds a new method to the interface for fetching all counties and their carbon intensity.

To add year as we're discussing in https://github.com/thegreenwebfoundation/grid-intensity-exporter/issues/7 we'll need to change the signature here.

But I'd prefer to do that as a follow up as these changes are already quite big.

Reply:
Yes, let's defer that decision.